grobid-core/src/main/java/org/grobid/core/data/Date.java

```diff
@@ -214,4 +214,19 @@ public String toXML() {
         return theDate;
     }
 
+    public String toISO() {
+        if (year == -1) {
+            return null;
+        }
+        String theDate = String.valueOf(year);
+        if (month == -1) {
+            return theDate;
+        }
+        theDate += "-" + String.valueOf(month);
+        if (day != -1) {
+            theDate += "-" + String.valueOf(day);
+        }
+        return theDate;
+    }
+
```

Comment:
Re-use: https://github.com/kermitt2/grobid/blob/master/grobid-core/src/main/java/org/grobid/core/document/TEIFormatter.java#L108
-> it manages correctly the 0-digit for match the ISO pattern.